ios/testmanagerd/xctestrunutils.go

```diff
@@ -0,0 +1,173 @@
+	testsToRun := data.TestConfig.OnlyTestIdentifiers
+	testsToSkip := data.TestConfig.SkipTestIdentifiers
+
+	testEnv := make(map[string]any)
+	maps.Copy(testEnv, data.TestConfig.EnvironmentVariables)
+	maps.Copy(testEnv, data.TestConfig.TestingEnvironmentVariables)
+
+	// Extract only the file name
+	var testBundlePath = filepath.Base(data.TestConfig.TestBundlePath)
+
+	// Build the TestConfig object from parsed data
+	testConfig := TestConfig{
+		TestRunnerBundleId: data.TestConfig.TestHostBundleIdentifier,
+		XctestConfigName:   testBundlePath,
+		Args:               data.TestConfig.CommandLineArguments,
+		Env:                testEnv,
+		TestsToRun:         testsToRun,
+		TestsToSkip:        testsToSkip,
+		XcTest:             !data.TestConfig.IsUITestBundle,
+		Device:             device,
+		Listener:           listener,
+	}
+
+	return testConfig, nil
+}
+
+// SchemeData represents the structure of a scheme-specific test configuration
+type SchemeData struct {
+	TestHostBundleIdentifier    string         `plist:"TestHostBundleIdentifier"`
+	TestBundlePath              string         `plist:"TestBundlePath"`
+	SkipTestIdentifiers         []string       `plist:"SkipTestIdentifiers"`
+	OnlyTestIdentifiers         []string       `plist:"OnlyTestIdentifiers"`
+	IsUITestBundle              bool           `plist:"IsUITestBundle"`
+	CommandLineArguments        []string       `plist:"CommandLineArguments"`
+	EnvironmentVariables        map[string]any `plist:"EnvironmentVariables"`
+	TestingEnvironmentVariables map[string]any `plist:"TestingEnvironmentVariables"`
+}
+
+// XCTestRunMetadata contains metadata about the .xctestrun file
+type XCTestRunMetadata struct {
+	FormatVersion int `plist:"FormatVersion"`
+}
+
+// XCTestRunCodec is a utility for parsing .xctestrun files with FormatVersion 1.
+// It extracts test configurations and metadata, providing a structured SchemeData object
+// that contains all the necessary information required to execute a test.
+// This includes details like test bundle paths, environment variables, command-line arguments,
+// and other configuration settings essential for running tests.
+
+// XCTestRunCodec handles encoding and decoding operations for .xctestrun files
+type XCTestRunCodec struct{}
+
+// NewXCTestRunCodec creates a new instance of XCTestRunCodec
+func NewXCTestRunCodec() XCTestRunCodec {
+	return XCTestRunCodec{}
+}
+
+// ParseFile reads the .xctestrun file and decodes it into a map
+func (codec XCTestRunCodec) ParseFile(filePath string) (XCTestRunData, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return XCTestRunData{}, fmt.Errorf("failed to open xctestrun file: %w", err)
+	}
+	defer file.Close()
+	return codec.Decode(file)
+}
+
+// Decode decodes the binary xctestrun content into the XCTestRunData struct
+func (codec XCTestRunCodec) Decode(r io.Reader) (XCTestRunData, error) {
+	// Read the entire content once
+	content, err := io.ReadAll(r)
```

Comment:
Not sure how go handles this, but we should make sure that we are not leaking any resources here and close the reader @dmissmann?

Reply:
Whoever creates the resource should close it. And this is happening in `ParseFile`, so that's fine

Reply:
cleanup will be ensured with `defer file.Close().`